activity/src/main/java/com/amazonaws/stepfunctions/cloudformation/activity/Constants.java

```diff
@@ -1,8 +1,20 @@
 package com.amazonaws.stepfunctions.cloudformation.activity;
 
+import com.google.common.collect.Sets;
+
+import java.util.Set;
+
 public class Constants {
     public static final String THROTTLING_ERROR_CODE = "ThrottlingException";
     public static final String ACCESS_DENIED_ERROR_CODE = "AccessDeniedException";
     public static final String INTERNAL_FAILURE_MESSAGE = "Internal Failure";
     public static final String ACTIVITY_DOES_NOT_EXIST_ERROR_CODE = "ActivityDoesNotExist";
+    public static final String RESOURCE_DOES_NOT_EXIST_ERROR_CODE = "ResourceNotFound";
+    public static final String INVALID_ARN_ERROR_CODE = "InvalidArn";
+
+    public static final Set<String> RESOURCE_NOT_FOUND_ERROR_CODES = Sets.newHashSet(
+            RESOURCE_DOES_NOT_EXIST_ERROR_CODE,
+            ACTIVITY_DOES_NOT_EXIST_ERROR_CODE,
+            INVALID_ARN_ERROR_CODE
```

Comment:
is this really a resource not found error code.

The message makes it seem like it's a validation error that I presume covers resource not found in addition to other circumstances like malformed and invalid ARNs. Is that the case?

If it's truly a resource not found / 404 type of error code, it should probably include some form of "not found" in it

Reply:
+1 .
I also didn't find the usage of RESOURCE_DOES_NOT_EXIST_ERROR_CODE is this a place holder for general resource errors?

Reply:
Refactored to use custom inputs to the contract tests, which removes the need to handle Invalid Arn error codes. The `RESOURCE_DOES_NOT_EXIST_ERROR_CODE` error code is the error code thrown from the TagResource/UntagResource API calls, so even though it's not referenced in the code, we need to include in the list of errors to translate to the NotFound handler error code.